Update text-generation page with tools, models and datasets (#900)
huggingface/hub-docs: tasks/src/text-generation/about.md

})
```

## Text Generation Inference

> [Text Generation Inference (TGI)](https://github.com/huggingface/text-generation-inference) is a toolkit developed by Hugging Face to address challenges on response time and latency for concurrent users in serving large language models. It's an open-source serving solution that powers inference solutions of Hugging Face, [Inference Endpoints](https://huggingface.co/inference-endpoints), and [Inference API](https://huggingface.co/docs/api-inference/index) and [Hugging Chat](https://huggingface.co/chat/). You can use it to deploy any open-source large language model of your choice.

Reviewer: ```suggestion
[Text Generation Inference (TGI)](https://github.com/huggingface/text-generation-inference) is an open-source toolkit for serving LLMs tackling challenges such as response time. TGI powers inference solutions like [Inference Endpoints](https://huggingface.co/inference-endpoints) and [Hugging Chat](https://huggingface.co/chat/), as well as multiple community projects. You can use it to deploy any supported open-source large language model of your choice.
```

## ChatUI Spaces

> Hugging Face Spaces includes templates to easily deploy your own instance of a specific application. [ChatUI](https://github.com/huggingface/chat-ui), is an open-source interface that enables serving conversational interface for large language models, and can be deployed with few clicks at Spaces. TGI powers these Spaces under the hood for faster inference. Thanks to the template, you can deploy your own instance based on a large language model with only a few clicks and customize it. Learn more about it [here](https://huggingface.co/docs/hub/spaces-sdks-docker-chatui) and create your large language model instance [here](https://huggingface.co/new-space?template=huggingchat/chat-ui-template).

Reviewer: ```suggestion
Hugging Face Spaces includes templates to easily deploy your own instance of a specific application. [ChatUI](https://github.com/huggingface/chat-ui) is an open-source interface that enables serving conversational interface for large language models and can be deployed with few clicks at Spaces. TGI powers these Spaces under the hood for faster inference. Thanks to the template, you can deploy your own instance based on a large language model with only a few clicks and customize it. Learn more about it [here](https://huggingface.co/docs/hub/spaces-sdks-docker-chatui) and create your large language model instance [here](https://huggingface.co/new-space?template=huggingchat/chat-ui-template).
```

> Hugging Face Spaces includes Docker templates to easily deploy your own instance of a specific application. [ChatUI](https://github.com/huggingface/chat-ui), is an open-source interface that enables serving conversational interface for large language models, and is currently available for Docker Spaces. The Docker template for ChatUI is powered by TGI for faster inference. Thanks to the template, you can deploy own instance based on a large language model with only a few clicks and customize it. Learn more about it [here](https://huggingface.co/docs/hub/spaces-sdks-docker-chatui) and create your large language model instance [here](https://huggingface.co/new-space?template=huggingchat/chat-ui-template).

Reviewer: This might have too much advanced jargon. E.g. I don't expect most of our audience to know what Docker or Spaces is. Maybe the language here can be made a bit more beginner-friendly

Author: I iterated on it yesterday, can you check again if possible?

![ChatUI](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/blog/os_llms/docker_chat.png)

## Useful Resources

Would you like to learn more about the topic? Awesome! Here you can find some curated resources that you may find helpful!

### Tools within Hugging Face Ecosystem
- You can use [PEFT](https://github.com/huggingface/peft) to adapt large language models in efficient way.

> - [ChatUI](https://github.com/huggingface/chat-ui) is the open-source interface built by Hugging Face to conversate with Large Language Models.

Reviewer: ```suggestion
- [ChatUI](https://github.com/huggingface/chat-ui) is the open-source interface to conversate with Large Language Models.
```
